src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java

```diff
@@ -139,44 +137,126 @@ public static Bitmap getBitmapFromDiskCache(String key) {
         return null;
     }
 
-    
-    public static boolean cancelPotentialWork(OCFile file, ImageView imageView) {
-        final ThumbnailGenerationTask bitmapWorkerTask = getBitmapWorkerTask(imageView);
+    public interface AsyncTaskFile {
```

Comment:
Adding ```AsyncTaskFile``` and the two classes implementing is resulting in a lot of extra lines without not so good reasons. 

Only the three first methods of the interface are useful. All the others are only called from ```ThumbnailGenerationTask.doInBackground```, inside a block conditioned by a casting of ```AsyncTaskFile``` to one of its subtypes -> that' s not taking advantage of polymorphism at all.

In this case I would forget interfaces and implementations. I'd fully remove this interface and its implementations. Check more comments on ```ThumbnailGenerationTask``` below to see how adapt it.